sdk/python/kfp/cli/diagnose_me/kubernetes_cluster.py

```diff
@@ -0,0 +1,122 @@
+# Lint as: python3
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License,Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions for collecting diagnostic information on Kubernetes cluster."""
+
+import enum
+from typing import List, Text, Optional
+from . import utility
+
+
+class Commands(enum.Enum):
+  """Enum for kubernetes commands."""
+  K8_CONFIGURED_CONTEXT = 1
+  PODS = 2
+  PVCS = 3
+  PVS = 4
+  SECRETS = 5
+  SERVICES = 6
+  KUBECTL_VERSION = 7
+
+
+_command_string = {
+    Commands.K8_CONFIGURED_CONTEXT: 'config view',
+    Commands.PODS: 'get pods',
+    Commands.PVCS: 'get pvc',
+    Commands.PVS: 'get pv',
+    Commands.SECRETS: 'get secrets',
+    Commands.SERVICES: 'get services',
+    Commands.KUBECTL_VERSION: 'version',
+}
+
+
+def execute_kubectl_command(
+    kubectl_command_list: List[Text],
+    human_readable: Optional[bool] = False) -> utility.ExecutorResponse:
+  """Invokes  the kubectl command.
+
+  Args:
+    kubectl_command_list: a command string list to be past to kubectl example
+      format is ['config', 'view']
+    human_readable: If false sets parameter -o json for all calls, otherwie
+      output will be in human readable format.
+
+  Returns:
+    utility.ExecutorResponse with outputs from stdout,stderr and execution code.
+  """
+  command_list = ['kubectl']
+  command_list.extend(kubectl_command_list)
+  if not human_readable:
+    command_list.extend(['-o', 'json'])
+
+  return utility.ExecutorResponse().execute_command(command_list)
+
+
+def get_kubectl_configuration(
+    configuration: Commands,
+    kubernetes_context: Text = None,
+    namespace: Text = None,
+    human_readable: Optional[bool] = False) -> utility.ExecutorResponse:
```

Comment:
Should this be "tri-state" or boolean?

Reply:
Ack, two state Boolean would be sufficient for now, though noted in future refactoring may need to revisit this flag.

Reply:
What I meant is this should probably be
```suggestion
    human_readable: bool = False) -> utility.ExecutorResponse:
```

Reply:
Done :) updated accordingly.